<launch>

    <!-- Add command line parameter for subject number -->
    <arg name="sub_number" />

    <!-- Select path to save the bag file to -->
    <arg name="output_path" default="$(find hri_predict_ros)/logs/bag/test_offline_simple_sub_$(arg sub_number)" />

    <!-- Select the topics to record -->
    <arg name="zed_skeleton_topic" default="/zed/zed_node/body_trk/skeletons" />
    <!-- <arg name="zed_image_topic" default="/zed/zed_node/rgb/image_rect_color" /> -->
    <!-- <arg name="zed_info_topic" default="/zed/zed_node/rgb/camera_info" /> -->
    <arg name="tf_topic" default="/tf" />
    <arg name="tf_static_topic" default="/tf_static" />

    <!-- Select the namespace to record topics to -->
    <arg name="offline_ns" default="/offline" />

    <!-- Call rosbag record to record the robot joint states and human keypoints + tf data -->
    <!-- <node name="record" pkg="rosbag" type="record" args="-O $(arg output_path)
                                                            $(arg offline_ns)$(arg zed_skeleton_topic)
                                                            $(arg offline_ns)$(arg zed_image_topic)
                                                            $(arg offline_ns)$(arg zed_info_topic)
                                                            $(arg offline_ns)$(arg tf_topic)
                                                            $(arg offline_ns)$(arg tf_static_topic)">
        <remap from="$(arg offline_ns)$(arg zed_skeleton_topic)" to="$(arg zed_skeleton_topic)" />
        <remap from="$(arg offline_ns)$(arg zed_image_topic)" to="$(arg zed_image_topic)" />
        <remap from="$(arg offline_ns)$(arg zed_info_topic)" to="$(arg zed_info_topic)" />
        <remap from="$(arg offline_ns)$(arg tf_topic)" to="$(arg tf_topic)" />
        <remap from="$(arg offline_ns)$(arg tf_static_topic)" to="$(arg tf_static_topic)" />
    </node> -->
    <node name="record" pkg="rosbag" type="record" args="-O $(arg output_path)
                                                            $(arg offline_ns)$(arg zed_skeleton_topic)
                                                            $(arg offline_ns)$(arg tf_topic)
                                                            $(arg offline_ns)$(arg tf_static_topic)">
        <remap from="$(arg offline_ns)$(arg zed_skeleton_topic)" to="$(arg zed_skeleton_topic)" />
        <remap from="$(arg offline_ns)$(arg tf_topic)" to="$(arg tf_topic)" />
        <remap from="$(arg offline_ns)$(arg tf_static_topic)" to="$(arg tf_static_topic)" />
    </node>
</launch>